### PR TITLE
Working on the `GLIBC_2.25 not found` build issue 

### PR DESCRIPTION
**Pull request by avsthiago**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by avsthiago** [comment]
Hi @xhochy, @janjagusch, and @jtilly,

I'm addressing a build issue in our conda feedstock ([link to build issue](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=826657&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=563)) due to CentOS 7's outdated GLIBC conflicting with Node 18 requirements. 

Since code-server [doesn't support cos7 anymore](https://github.com/coder/code-server/issues/6451#issuecomment-1737738997), would building the package on a different Linux distribution with a newer GLIBC be a feasible workaround in the conda-forge context? Any guidance or alternative suggestions would be greatly appreciated


```
+ code-server --help
node:internal/modules/cjs/loader:1340
  return process.dlopen(module, path.toNamespacedPath(filename));
                 ^

Error: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /home/conda/feedstock_root/build_artifacts/code-server_1701001419227/_h_env_placehold/share/code-server/node_modules/argon2/lib/binding/napi-v3/argon2.node)
    at Module._extensions..node (node:internal/modules/cjs/loader:1340:18)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12)
    at Module.require (node:internal/modules/cjs/loader:1143:19)
    at require (node:internal/modules/cjs/helpers:119:18)
    at Object.<anonymous> (/home/conda/feedstock_root/build_artifacts/code-server_1701001419227/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/share/code-server/node_modules/argon2/argon2.js:9:25)
    at Module._compile (node:internal/modules/cjs/loader:1256:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
    at Module.load (node:internal/modules/cjs/loader:1119:32)
    at Module._load (node:internal/modules/cjs/loader:960:12) {
  code: 'ERR_DLOPEN_FAILED'

```

**Comment by xhochy** [comment]
At the moment, it's not possible.

**Comment by avsthiago** [comment]
@xhochy, what do you think about this solution? I kept the `code-server --help` test for the `osx` and `aarch64` and reduced the test to only `which code-server` when building the `linux64` version. 

My goal here is to unblock the `osx` and `aarch64` builds. I also expect the package to only fail on Linux only if the system doesn't have GLIBC > 2.25.

**Review by xhochy — CHANGES_REQUESTED** [review]
I would prefer to delay this until we have AlmaLinux support. That is probably 1-2 months away.

Until then, this should at least have a different `__glibc` (?) pin.

**Comment by avsthiago** [comment]
@xhochy, could you please review this PR again when you have time?

File: .ci_support/linux_64_.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ cxx_compiler:
 cxx_compiler_version:
 - '12'
 docker_image:
-- quay.io/condaforge/linux-anvil-cos7-x86_64
+- quay.io/condaforge/linux-anvil-alma-x86_64:8
```
Comment (xhochy) [inline .ci_support/linux_64_.yaml:12]: Please don't manually touch these files. You will need to increase the `c_stdlib_version` for this to work.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -1,2 +1,3 @@
-c_stdlib_version:              # [linux]
-  - 2.17                       # [linux]
\ No newline at end of file
+# use a docker image with glibc 2.28 in it for testing
+docker_image:                                  # [x86_64 and linux64]
```
Comment (xhochy) [inline recipe/conda_build_config.yaml:2]: As said before, this is should be done via `c_stdlib_version` and not the docker image directly.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: .ci_support/linux_64_.yaml (hunk 1 above)
Comment (avsthiago) [reply .ci_support/linux_64_.yaml]: I reverted this change.

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (avsthiago) [reply recipe/conda_build_config.yaml]: I reverted this change too. I tried to follow the suggestions from https://conda-forge.org/news/2024/03/24/stdlib-migration/ but no success so far. 

Do you know where would be the right place for me to place the `c_stdlib_version`?

Currently I'm getting the following error in the build.

```
 conda_libmamba_solver.conda_build_exceptions.ExplainedDependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("c_linux-64=2.17")}
    Encountered problems while solving:
      - nothing provides requested c_linux-64 2.17.*
    
    Could not solve for environment specs
    The following package could not be installed
    └─ c_linux-64 2.17.*  does not exist (perhaps a typo or a missing channel).
```

Comment (xhochy) [reply recipe/conda_build_config.yaml]: You need to rerender the recipe for this to take effect.

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

**Comment by avsthiago** [comment]
@conda-forge-admin please rerender

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (avsthiago) [reply recipe/conda_build_config.yaml]: I didn't know I should do that, thanks! 

I updated `c_stdlib_version` from 2.17 to 2.28 and run a rerender, but it still gives me the original error. 

I expected that after updating to 2.28 and running a rerender the file `.ci_support/linux_64_.yaml` would get updated to use the image `quay.io/condaforge/linux-anvil-alma-x86_64:8`. Is that the right assumption?

fyi: I previously set the docker image directly based the implementation done at the [onednn-feedstock](https://github.com/conda-forge/onednn-feedstock/blob/main/recipe/conda_build_config.yaml#L3) repo

**Comment by ehfd** [comment]
```
c_stdlib:                  # [linux]
  - sysroot                # [linux]
c_stdlib_version:          # [linux]
  - "2.28"                 # [linux]
```

I change changing the `conda_build_config.yaml` to this format. This way, `sysroot` is the `c_stdlib`.

**Comment by ehfd** [comment]
@avsthiago Unfortunately, c_stdlib as sysroot was probably not it. After rerender, the logs changed to something else.

**Comment by avsthiago** [comment]
Yeah, @ehfd, that didn't work. I'll give it another try with the Alma image.

**Comment by avsthiago** [comment]
@conda-forge-admin please rerender

**Comment by ehfd** [comment]
Looks like the original solutions were right, except needing to rerender after putting it into conda_build_config.yaml.

**Comment by avsthiago** [comment]
@xhochy I tried to fix the build by only changing the `c_stdlib_version`, but that didn't work. The only fix I found so far was to change the `docker_image`. 

Do you see any other option?

**Comment by ehfd** [comment]
I guess changing docker_image for conda_build_config.yaml and rerendering is OK, if not changing the .script folder?

**Comment by xhochy** [comment]
`c_stdlib_version` is the way to go. What was the error? 

Otherwise the version constraints will not be set correctly.

**Comment by ehfd** [comment]
@xhochy 

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=917329&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=917329&view=logs&jobId=6494c7a6-a019-59a3-6f3d-9bab8ca96717&j=6494c7a6-a019-59a3-6f3d-9bab8ca96717&t=18c91bca-a25a-5f89-34d4-93bbec4021bc

Compare the two. linux_64 still shows `Error: /lib64/libc.so.6: version `GLIBC_2.25' not found` for testing.

**Comment by ehfd** [comment]
@avsthiago @xhochy Perhaps require stdlib('c') with 2.28 constraint in the test environment as well?

**Comment by xhochy** [comment]
We probably need the docker image and `c_stdlib_version` at the same time. In future, setting the docker image should be obsolete, but until the new sysroot is fully rolled out, we will need to add that to `conda_build_config.yaml`, too.

**Comment by ehfd** [comment]
@xhochy Then everything is as it is now.

**Review by xhochy — APPROVED** [review]

**Pull request merged**